streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -322,14 +323,16 @@ void setConsumer(final Consumer<byte[], byte[]> consumer) {
      * @throws StreamsException if the store's change log does not contain the partition
      */
     boolean updateNewAndRestoringTasks() {
+        // wait for other thread in the same jvm to free the locks of state store
+        Utils.sleep(10);
```

Comment:
This can be taken out.

When the `StreamThread` goes into the `runOnce` method and the state is `PARTITIONS_ASSIGNED` the `StreamThread` calls `taskManager.updateNewAndRestoringTasks()`.  If the lock is held by another thread still cleaning up this call will return false and the state will not transition to running.  In the next iteration of `runOnce()` the `StreamThread` will attempt the same operation again since the state has not yet been set to `RUNNING`

Reply:
Thanks for your detailed explanation, there is indeed no need for waiting. `StreamThread.runLoop` will keep retry until the lock is free and `taskManager.updateNewAndRestoringTasks()` return true.

Btw, is it possible that the lock can not be free forever ?